src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java

```diff
@@ -0,0 +1,119 @@
+package gregtech.api.capability.impl;
+
+import gregtech.api.GTValues;
+import gregtech.api.capability.IMultipleTankHandler;
+import gregtech.api.recipes.Recipe;
+import gregtech.api.recipes.RecipeMap;
+import net.minecraftforge.fluids.FluidTank;
+import net.minecraftforge.fluids.IFluidTank;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+public class SteamMultiblockRecipeLogic extends AbstractRecipeLogic {
+
+    private IMultipleTankHandler steamFluidTank;
+    private IFluidTank steamFluidTankCombined;
+
+    // EU per mB
+    private final double conversionRate;
+
+    public SteamMultiblockRecipeLogic(RecipeMapSteamMultiblockController tileEntity, RecipeMap<?> recipeMap, IMultipleTankHandler steamFluidTank, double conversionRate) {
+        super(tileEntity, recipeMap);
+        this.steamFluidTank = steamFluidTank;
+        this.conversionRate = conversionRate;
+        allowOverclocking = false;
+        combineSteamTanks();
+    }
+
+    public IFluidTank getSteamFluidTankCombined() {
+        combineSteamTanks();
+        return steamFluidTankCombined;
+    }
+
+    @Override
+    protected IItemHandlerModifiable getInputInventory() {
+        RecipeMapSteamMultiblockController controller = (RecipeMapSteamMultiblockController) metaTileEntity;
+        return controller.getInputInventory();
+    }
+
+    @Override
+    protected IItemHandlerModifiable getOutputInventory() {
+        RecipeMapSteamMultiblockController controller = (RecipeMapSteamMultiblockController) metaTileEntity;
+        return controller.getOutputInventory();
+    }
+
+    protected IMultipleTankHandler getSteamFluidTank() {
+        RecipeMapSteamMultiblockController controller = (RecipeMapSteamMultiblockController) metaTileEntity;
+        return controller.steamFluidTank;
+    }
+
+    private void combineSteamTanks() {
+        steamFluidTank = getSteamFluidTank();
+        if (steamFluidTank == null)
+            steamFluidTankCombined = new FluidTank(0);
+        else {
+            int capacity = steamFluidTank.getTanks() * 64000;
+            steamFluidTankCombined = new FluidTank(capacity);
+            steamFluidTankCombined.fill(steamFluidTank.drain(capacity, false), true);
+        }
+    }
+
+    @Override
+    public void update() {
+
+        // Fixes an annoying GTCE bug in AbstractRecipeLogic
```

Comment:
you could fix the bug mentioned here :wink:

Reply:
> you could fix the bug mentioned here 😉
I will do that in another PR since I don't know if it will break other things.

Reply:
Ok, good idea